File: older/gae-foodtruck-api/src/main/java/com/example/foodtruck/services_implementation/IngredientServiceImplementation.java
package com.example.foodtruck.services_implementation;

import com.example.foodtruck.exceptions.EntityNotFoundException;
import com.example.foodtruck.models.Ingredient;
import com.example.foodtruck.repositories.IngredientRepository;
import com.example.foodtruck.services.IngredientService;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.Optional;

import static java.util.Arrays.asList;

@Service
public class IngredientServiceImplementation implements IngredientService {

    private IngredientRepository repository;

    @Autowired
    public IngredientServiceImplementation(IngredientRepository repository){
        this.repository = repository;
    }

    @Override
    public Ingredient findById(Long id) {
        Optional<Ingredient> ingredientObject = repository.findById(id);
        return ingredientObject.orElseThrow(EntityNotFoundException::new);
    }

    @Override
    public Iterable<Ingredient> getAll(){
        return repository.findAll();
    }

    @Override
    public Iterable<Ingredient> save(Ingredient ingredient){
        return repository.saveAll(asList(ingredient));
    }

    @Override
    public Iterable<Ingredient> update(Long id, Ingredient ingredient){
        return repository.saveAll(asList(ingredient));
    }

    @Override
    public void delete(Long id){
        this.findById(id);
        repository.deleteById(id);
    }
}

File: api/src/main/kotlin/com/foodtruck/api/services_implementation/IngredientServiceImplementation.kt
package com.foodtruck.api.services_implementation

import com.foodtruck.api.exceptions.EntityNotFoundException
import com.foodtruck.api.models.Ingredient
import com.foodtruck.api.repositories.IngredientRepository
import com.foodtruck.api.services.IngredientService

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

import java.util.Arrays.asList

@Service
class IngredientServiceImplementation @Autowired
constructor(private val repository: IngredientRepository) : IngredientService {

    override fun findById(id: Long): Ingredient {
        val ingredientObject = repository.findById(id)
        return ingredientObject.orElseThrow<EntityNotFoundException>(::EntityNotFoundException)
    }

    override fun getAll(): Iterable<Ingredient> {
        return repository.findAll()
    }

    override fun save(ingredient: Ingredient): Iterable<Ingredient> {
        return repository.saveAll(asList(ingredient))
    }

    override fun update(id: Long, ingredient: Ingredient): Iterable<Ingredient> {
        return repository.saveAll(asList(ingredient))
    }

    override fun delete(id: Long) {
        this.findById(id)
        repository.deleteById(id)
    }
}
